src/main/java/xyz/nucleoid/bedwars/game/active/shop/BwItemShop.java

```diff
@@ -6,110 +6,200 @@
 import xyz.nucleoid.bedwars.game.active.upgrade.PlayerUpgrades;
 import xyz.nucleoid.bedwars.game.active.upgrade.Upgrade;
 import xyz.nucleoid.bedwars.game.active.upgrade.UpgradeType;
+import xyz.nucleoid.bedwars.util.BwPotions;
 import xyz.nucleoid.plasmid.shop.Cost;
 import xyz.nucleoid.plasmid.shop.ShopBuilder;
 import xyz.nucleoid.plasmid.shop.ShopEntry;
 import xyz.nucleoid.plasmid.shop.ShopUi;
 import xyz.nucleoid.plasmid.util.ColoredBlocks;
 import xyz.nucleoid.plasmid.util.ItemStackBuilder;
+
+import java.util.function.BiFunction;
+
 import net.minecraft.block.Blocks;
+import net.minecraft.enchantment.Enchantment;
 import net.minecraft.enchantment.Enchantments;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemConvertible;
 import net.minecraft.item.ItemStack;
 import net.minecraft.item.Items;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.potion.PotionUtil;
+import net.minecraft.potion.Potions;
 import net.minecraft.server.network.ServerPlayerEntity;
 import net.minecraft.text.LiteralText;
 import net.minecraft.text.Text;
 import net.minecraft.util.DyeColor;
 
 public final class BwItemShop {
-    //TODO: reorganize shop layout
     public static ShopUi create(ServerPlayerEntity player, BwActive game) {
-        return ShopUi.create(new LiteralText("Item Shop"), shop -> {
+        //TODO User preferred items page
```

Comment:
is this going to be implemented in this PR, or is it non-blocking

Reply:
Seperate PR, not done yet but no affect to the end user